proposals/csharp-7.0/tuples.md

```diff
@@ -290,37 +311,37 @@ System.ValueTuple<int, int> vt = t;  // identity conversion
 
+\[Example:
 ```csharp
 (string first, string last) GetNames() { ... }
 (string last, string first) names = GetNames(); // Oops!
 ```
-
-> Compilers should issue a warning for the preceding code.
->
-> endnote.
+end example\]
 
 ### Boxing conversions
 
-> The following text should be added to [Boxing conversions](../../spec/conversions.md#boxing-conversions) after the first paragraph:
+> Add the following text to [Boxing conversions](../../spec/conversions.md#boxing-conversions) after the first paragraph:
 
-Tuples, like all value types, have a boxing conversion. Importantly, the names aren't part of the runtime representation of tuples, but are tracked only by the compiler. Thus, once you've "cast away" the names, you cannot recover them. In alignment with the identity conversions, a boxed tuple will unbox to any tuple type that has the same element types in the same order.
+Tuples have a boxing conversion. Importantly, the element names aren't part of the runtime representation of tuples, but are tracked only by the compiler. Thus, once element names have been "cast away", they cannot be recovered. In alignment with identity conversion, a boxed tuple unboxes to any tuple type that has the same element types in the same order.
 
 ### Tuple conversions
 
-> This section should be added after [Implicit enumeration conversions](../../spec/conversions.md#implicit-enumeration-conversions)
+> Add this section after [Implicit enumeration conversions](../../spec/conversions.md#implicit-enumeration-conversions)
 
-Tuple types and expressions support a variety of conversions by "lifting" conversions of the elements into overall *tuple conversion*.
-For the classification purpose, all element conversions are considered recursively. For example: To have an implicit conversion, all element expressions/types must have implicit conversions to the corresponding element types.
+Tuple types and expressions support a variety of conversions by "lifting" conversions of the elements into overall *tuple conversion*. For the classification purpose, all element conversions are considered recursively. For example, to have an implicit conversion, all element expressions/types shall have implicit conversions to the corresponding element types.
 
 Tuple conversions are *Standard Conversions* and therefore can stack with user-defined operators to form user-defined conversions.
 
+**ISSUE:** Is 'stack' a defined term in this context?
+
 An implicit tuple conversion is a standard conversion. It applies between two tuple types of equal arity when there is any implicit conversion between each corresponding pair of types.
 
+**ISSUE:** Re 'arity' does this mean number of elements, with their types and names? In any event, perhaps we should define this in terms of tuples and consistently use it throughout, as in, "The arity of a tuple is ...".
```

Comment:
*arity* means number of elements and their types. I don't think it includes names. We should verify that by looking at all uses.

Reply:
I defined "arity" in Tuple types|General, as follows:

>A tuple's ***arity*** is the combination of its element types, in lexical order; element names are *not* included.

and I reviewed the proposal, consistently using this term, as appropriate.

Having said that, my experience with using "tuple" has to do with counts only: the number of operands an operator takes or the number of arguments a function expects; type info does not enter into it. Yet, in the case of tuples, element type is significant. As such, like functions/methods, tuples really have "signatures", which capture both number and ordered type. Might we be able to use that term instead?